.gitlab-ci.yml

```
@@ -858,13 +861,16 @@ changelog:
         --target-branch=${CI_COMMIT_REF_NAME}
     # git cliff: override the changelog
     - test $GIT_CLIFF == "false" && echo "INFO - Skipping git-cliff" && exit 0
-    - git clone https://${GITHUB_USER}:${GITHUB_BOT_TOKEN_REPO_FULL}@github.com/${GITHUB_REPO_URL}
-    - cd ${GITHUB_REPO_URL#*/}
+    - git remote add github https://${GITHUB_USER}:${GITHUB_BOT_TOKEN_REPO_FULL}@github.com/${GITHUB_REPO_URL}
     - RELEASE_PLEASE_PR=$(gh pr list --author "${GITHUB_USER_NAME}" --head "release-please--branches--${CI_COMMIT_REF_NAME}" --json number | jq -r '.[0].number // empty')
+    - gh repo set-default https://${GITHUB_USER}:${GITHUB_BOT_TOKEN_REPO_FULL}@github.com/${GITHUB_REPO_URL}
     - test -z "$RELEASE_PLEASE_PR" && echo "No release-please PR found" && exit 0
     - echo "INFO - generating changelog notes from git cliff"
         && gh pr checkout $RELEASE_PLEASE_PR
-        && git cliff --unreleased --output mender/CHANGELOG.md --bump --github-repo ${GITHUB_REPO_URL}
+        && new_changelog=$(git cliff --unreleased --bump --github-repo ${GITHUB_REPO_URL} | sed 's/---//')
+        && echo $new_changelog | cat - mender/CHANGELOG.md > tmp_changelog.md
+        && echo "${CHANGELOG_HEADER}\n" > mender/CHANGELOG.md
+        && cat tmp_changelog.md | grep -v ${CHANGELOG_HEADER} >> mender/CHANGELOG.md
```

What about using `--prepend` instead of `--output`?
```suggestion
        && git cliff --unreleased --prepend mender/CHANGELOG.md --bump --github-repo ${GITHUB_REPO_URL}
```

---

Yeah! Much nicer!

---

I still have to update the header `# Mender Helm chart` and remove the `---`, but it's better than before.